mobtravelapp/src/main/java/com/mobiquity/mobtravelapp/service/TravelService.java

```
@@ -0,0 +1,248 @@
+        return new Trip(routeModelAfterReformat.getFromStation(), routeModelAfterReformat.getToStation(), routeModelAfterReformat.getDateTime(), extractAllRoutes(trips));
+
+    }
+
+    /**
+     * Extracts all trips from reading the result from ns.nl
+     *
+     * @param result a String of trips
+     * @return a JsonArray of trips
+     */
+    public JsonArray extractAllTrips(String result) {
+        JsonObject jsonObject = new JsonParser().parse(result).getAsJsonObject();
+        JsonArray trips = jsonObject.getAsJsonArray("trips");
+        return trips;
+    }
+
+    /**
+     * Extract all routes from trips
+     *
+     * @param trips a JsonArray of trips
+     * @return List of Routes
+     */
+    public List<Route> extractAllRoutes(JsonArray trips) {
+        List<Route> routes = new ArrayList<>();
+        AtomicInteger index = new AtomicInteger(1);
+        IntStream.range(0, trips.size()).mapToObj(i -> trips.get(i).getAsJsonObject()).forEach(trip -> {
+            if (!trip.get("status").getAsString().equals("NORMAL")) {
+                Route route = Route.builder()
+                        .index(index.getAndIncrement())
+                        .status(trip.get("status").getAsString())
+                        .build();
+
+                routes.add(route);
+            } else {
+                Route route = null;
+                try {
+                    route = Route.builder()
+                            .index(index.getAndIncrement())
+                            .plannedDurationInMinutes(trip.get("plannedDurationInMinutes").getAsInt())
+                            .transfers(trip.get("transfers").getAsInt())
+                            .status(trip.get("status").getAsString())
+                            .legs(extractAllLegs(trip.get("legs").getAsJsonArray()))
+                            .build();
+                } catch (WeatherException e) {
+                    e.printStackTrace();
```

don't swallow exceptions.

---

Or if you must because you do not care about them log clearly what happened.